bt-core/src/main/java/bt/torrent/data/DefaultDataWorker.java

```diff
@@ -58,15 +60,17 @@ public DefaultDataWorker(IRuntimeLifecycleBinder lifecycleBinder,
         this.verifier = verifier;
         this.blockCache = blockCache;
 
-        this.executor = Executors.newSingleThreadExecutor(new ThreadFactory() {
+        this.executor = new ThreadPoolExecutor(config.getDataWorkerCorePoolSize(),
```

Comment:
Have you verified that the futures are thread-safe? I'm trying to recall why I ended up having a single-threaded data worker, but can't..

Reply:
I am not 100% sure they are thread-safe.  
I just did some tests about my own using cases, though these test passes, but they may not enough.   
I will do more tests later and recommit again.

Reply:
What platform are you testing on? (x86, arm, ...)

Reply:
x86